packages/@sanity/cli/src/util/createTelemetryStore.ts

```diff
@@ -0,0 +1,174 @@
+
+const FIVE_MINUTES = 1000 * 60 * 5
+
+export const TELEMETRY_CONSENT_CONFIG_KEY = 'telemetryConsent'
+
+function isTrueish(value: string | undefined) {
+  if (value === undefined) return false
+  if (value.toLowerCase() === 'true') return true
+  if (value.toLowerCase() === 'false') return false
+  const number = parseInt(value, 10)
+  if (isNaN(number)) return false
+  return number > 0
+}
+
+const VALID_STATUSES = ['granted', 'denied', 'unset'] as const satisfies Readonly<ConsentStatus[]>
+type ValidConsentStatus = (typeof VALID_STATUSES)[number]
+
+function isValidConsentStatus(status: string): status is ValidConsentStatus {
+  return VALID_STATUSES.includes(status as ValidConsentStatus)
+}
+
+function parseConsent(value: string): ValidConsentStatus {
+  const normalizedValue = value.toLowerCase()
+  if (isValidConsentStatus(normalizedValue)) {
+    return normalizedValue
+  }
+  throw new Error(
+    'Invalid value provided for environment variable MOCK_CONSENT. Must be either "0", "1", "true", "false", "granted", "denied" or "unset"',
+  )
+}
+
+function createTelemetryClient(token: string) {
+  return createClient({
+    apiVersion: '2023-10-22',
+    token,
+    useProjectHostname: false,
+    useCdn: false,
+  })
+}
+
+let _client: SanityClient | null = null
+function getCachedClient(token: string) {
+  if (!_client) {
+    _client = createTelemetryClient(token)
+  }
+  return _client
+}
+
+interface Env {
+  DO_NOT_TRACK?: string
+  MOCK_CONSENT?: string
+}
+
+interface Options {
+  env: Env
+}
+
+export type ConsentInformation =
+  | {
+      status: Extract<ConsentStatus, 'granted'>
+      reason?: never
+    }
+  | {
+      status: Extract<ConsentStatus, 'undetermined'>
+      reason: 'unauthenticated' | 'fetchError'
+    }
+  | {
+      status: Extract<ConsentStatus, 'denied'>
+      reason?: 'localOverride'
+    }
+  | {
+      status: Extract<ConsentStatus, 'unset'>
+      reason?: never
+    }
+
+export function resolveConsent({env}: Options): Promise<ConsentInformation> {
+  debug('Resolving consent…')
+
+  if (isCi) {
+    debug('CI environment detected, treating telemetry consent as denied')
+    return Promise.resolve({status: 'denied'})
+  }
+
+  if (isTrueish(env.DO_NOT_TRACK)) {
+    debug('DO_NOT_TRACK is set, consent is denied')
+    return Promise.resolve({
+      status: 'denied',
+      reason: 'localOverride',
+    })
+  }
+
+  const token = getCliToken()
+  if (!token) {
+    debug('User is not logged in, consent is undetermined')
+    return Promise.resolve({
+      status: 'undetermined',
+      reason: 'unauthenticated',
+    })
+  }
+
+  const client = getCachedClient(token)
+
+  function fetchConsent() {
+    const telemetryConsentConfig = createExpiringConfig<{
+      status: ValidConsentStatus
+    }>({
+      store: getUserConfig(),
+      key: TELEMETRY_CONSENT_CONFIG_KEY,
+      ttl: FIVE_MINUTES,
+      fetchValue: () => client.request({uri: '/users/me/consents/telemetry'}),
```

Comment:
`vX/intake/telemetry-status` is online on staging

Reply:
what is the difference between these two endpoints? the there a benefit of using one over the other?

Reply:
This keeps all the backend telemetry calls to one service and also lets us control responses outside of the user store

Reply:
Updated to use `/intake/telemetry-status`